lib/src/util/childUtil.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, {ReactNode} from 'react';
+
+export function addPropsToChildren(children: ReactNode, props: object) {
+  return React.Children.map(children, (child: ReactNode) => {
+    if (React.isValidElement(child)) {
+      return React.cloneElement(child, props);
+    }
+    return child;
+  });
+}
+
+export function only(children: ReactNode, type: React.FC): React.ReactElement | null {
```

Comment:
Is `React.FC` the right type here?  I would've thought it would be `React.ComponentType`

Reply:
I'll switch to React.ComponentType